src/lsm/forest.zig

```diff
@@ -921,6 +956,9 @@ fn CompactionPipelineType(comptime Forest: type, comptime Grid: type) type {
             const first_beat = compaction_beat == 0;
             const half_beat = compaction_beat == @divExact(constants.lsm_batch_multiple, 2);
 
+            if (self.forest == null) self.forest = forest;
```

Comment:
Forest does `bar_apply_to_manifest` from `compact_callback`, so I figured it'd be simpler to just always have forest set. We still have `callback` to assert the proper lifecycle. 

Though, I do not like re-entrancy of `Forest.compact_callback`, would be good to fix that, but out of scope for this PR.

Reply:
> Though, I do not like re-entrancy of Forest.compact_callback, would be good to fix that, but out of scope for this PR.

Yes - that was originally to get things working with the existing manifest compaction without having to touch that code too!